src/dao/MoodDao.java

```diff
@@ -31,7 +41,41 @@ public MoodDao() {
      * executing the query
      */
     public ArrayList<Mood> getAllMoods() throws DaoException {
-        return null;
+        BufferedReader reader = null;
+        try {
+            ArrayList<Mood> moods = new ArrayList<Mood>();
+            String ln = "";
+
+            reader = new BufferedReader(new FileReader("data/Moods.csv"));
+            reader.readLine();
+
+            while ((ln = reader.readLine()) != null) {
+                String[] obj = ln.split(COMMA_DELIMITER);
+                if (obj.length > 0) {
+                    Mood md = new Mood(Integer.parseInt(obj[MOOD_ID_INDEX]), obj[MOOD_NAME_INDEX]);
+                    moods.add(md);
+                }
+
+                //Print Moods
+//                for (Mood md : moods) {
+//                    System.out.println(md.toString());
+//                }
+
+            }
+            return moods;
+            //TestHarness.getInstance().testMoodControllerRead(moods);
+
+        } catch (Exception e) {
+            System.out.println("Error in reading CSV file");
+            e.printStackTrace();
+        } finally {
+            try {
+                reader.close();
```

Comment:
I always forget this, but you need to check to see if the reader is null before closing it. If it is, then an avoidable exception will be thrown in the finally block. 

    if (reader != null) {
        reader.close();
    }

You could also look at try-with-resources to make file reading simpler, but your existing implementation is fine, too. There's nothing wrong with it. https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html